### PR TITLE
SVG Generator documentation

### PR DESCRIPTION
**Pull request by sosa-vaadin**
Documentation to install and use the new SVG Generator for Vaadin charts.

### CODE REVIEW

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: Exporting Charts as SVG
+order: 9
+layout: page
+---
+
+[[charts.svggenerator]]
+= Exporting Charts as SVG
+
+Charts can be exported as SVG by using the `SVGGenerator` class.
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:10]:
```suggestion
Charts can be exported as SVG by using the [classname]#SVGGenerator# class.
```

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+This allows you to generate an SVG string of any chart, using a `Configuration` object with the chart's data.
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:11]:
```suggestion
This allows you to generate an SVG string of any chart, using a [classname]#Configuration# object with the chart's data.
```

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+== Installing
+
+Add the `vaadin-charts-flow-svg-generator` dependency to your project's `pom.xml`.
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:15]:
```suggestion
Add the `vaadin-charts-flow-svg-generator` dependency to your project's `pom.xml` as follows:
```

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    <artifactId>vaadin-charts-flow-svg-generator</artifactId>
+</dependency>
+----
+
+.NodeJS required
+NOTE: NodeJS must be installed for the generator to work.
+
+== Using the Generator
+
+Create an instance of `SVGGenerator` and call the `generate()` method, passing the chart's `Configuration` as an argument.
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:30]:
```suggestion
Create an instance of [classname]#SVGGenerator# and call the [methodname]#generate()# method, passing the chart's [classname]#Configuration# as an argument.
```

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+The method returns a string with the SVG data of the chart.
+
+Remember to close the generator when you're done using it.
+It is recommended to use a `try-with-resources` block so the generator is automatically closed.
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:34]:
```suggestion
It is recommended to use a try-with-resources block, so the generator is automatically closed.
```

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+// ...
+try (SVGGenerator generator = new SVGGenerator()) {
+    String svg = generator.generate(configuration);
+}
+----
+
+== Customizing SVG Generation
+
+Some attributes about the resulting SVG can be customized, like width, height, theme, or language.
+Any customizations are handled with the `ExportOptions` class.
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:48]:
```suggestion
Any customizations are handled with the [classname]#ExportOptions# class.
```

File: articles/ds/components/charts/java-api/svg-generator.asciidoc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+options.setWidth(800);
+options.setHeight(600);
+try (SVGGenerator generator = new SVGGenerator()) {
+    String svg = generator.generate(configuration, options);
+}
+----
+
+=== Executing JavaScript Functions
+
+It is possible to execute functions from `Configuration` objects (for example: formatter functions).
```
Comment (magi42) [inline articles/ds/components/charts/java-api/svg-generator.asciidoc:64]:
```suggestion
It is possible to execute functions from [classname]#Configuration# objects, for example, formatter functions.
```

**Review by magi42 — CHANGES_REQUESTED** [review]

**Review by magi42 — APPROVED** [review]

**Comment by magi42** [comment]
Note that you can commit multiple suggestions in batch in the **Files changed** view.

**Comment by sosa-vaadin** [comment]
> Note that you can commit multiple suggestions in batch in the **Files changed** view.

I was unaware of it. Thanks!

**Comment by sosa-vaadin** [comment]
Waiting for SVG Generator feature release to merge this PR.

**Pull request merged**